### DISCUSSION TITLE
enable subtitles for all but two languages

### DISCUSSION THREAD

**Discussion by ghost** [CogentRedTester/mpv-sub-select#25] [Q&A] [answered]
There's no specific example [in the wiki](https://github.com/CogentRedTester/mpv-sub-select/wiki/Example-Configs).

As an example, a bilingual speaker (of English and German) wants to enable subtitles **for all but the said languages** (`en`, `de`). For all the other languages, set English subtitles (preferred) -- if not available, fall back to German.

This in `sub-select.json` doesn't appear to work:

```
[
    {
        "alang": [ "en?", "de?" ],
        "slang": "no"
    },
    {
        "alang": "*",
        "slang": [ "en?", "de?", "und" ]
    }
]
```

**note**: `"und"` means an undefined language tag

**Comment by ghost** [comment]
I didn't find an answer, but ~~I assume that the `?` character in Lua expands into other characters.~~ Sorry for any format errors.

**Reply by CogentRedTester** [reply, marked as answer]
No, `?` makes the previous character optional, so you're matching `en` and `e`. If you look at the top of that wiki page you can see it says that the 2 letter codes are used for DVDs, but presumably you're playing a digital file, in which case they use 3 letter codes, which you are not matching to.

**Comment by ghost** [comment]
Works as expected now after some testing with various media files. Some points worth noting:  

* I've commented out the `alang` option in my mpv config -- I expect that the first audio track is the original (i.e. non-dubbed track, which I never want to use)
* likewise for the `slang` setting
* I ran tests with the mpv's `--no-audio` flag which naturally conflicts with the extension as there's no audio track selected — use `--mute` instead (
